src/test/java/com/ryanharter/autoparcel/AutoParcelExtensionTest.java

```diff
@@ -88,4 +148,135 @@
+
+  private static class TestContext implements AutoValueExtension.Context {
+
+    private final ProcessingEnvironment processingEnvironment;
+    private final String packageName;
+    private final TypeElement autoValueClass;
+    private final Map<String, ExecutableElement> properties;
+
+    public TestContext(ProcessingEnvironment processingEnvironment, String packageName,
+                       TypeElement autoValueClass, Map<String, ExecutableElement> properties) {
+      this.processingEnvironment = processingEnvironment;
+      this.packageName = packageName;
+      this.autoValueClass = autoValueClass;
+      this.properties = properties;
+    }
+
+    public ProcessingEnvironment processingEnvironment() {
+      return processingEnvironment;
+    }
+
+    public String packageName() {
+      return packageName;
+    }
+
+    public TypeElement autoValueClass() {
+      return autoValueClass;
+    }
+
+    public Map<String, ExecutableElement> properties() {
+      return properties;
+    }
+  }
+
+  abstract class SampleTypeWithNonSerializable implements Parcelable {
+    abstract int primitive();
+    abstract String serializable();
+    abstract NonSerializable nonSerializable();
+  }
+
+  abstract class SampleTypeWithParcelable implements Parcelable {
+    abstract int primitive();
+    abstract String serializable();
+    abstract ParcelableProperty parcelable();
+  }
+
+  abstract class ParcelableProperty implements Parcelable {}
+
+  class NonSerializable {}
+
+  public static PackageElement getPackage(Element element) {
+    while(element.getKind() != ElementKind.PACKAGE) {
+      element = element.getEnclosingElement();
+    }
+
+    return (PackageElement)element;
+  }
+
+  public static ImmutableSet<ExecutableElement> getLocalAndInheritedMethods(TypeElement type, Elements elementUtils) {
+    LinkedHashMultimap methodMap = LinkedHashMultimap.create();
+    getLocalAndInheritedMethods(getPackage(type), type, methodMap);
+    LinkedHashSet overridden = new LinkedHashSet();
+    Iterator methods = methodMap.keySet().iterator();
+
+    while(methods.hasNext()) {
+      String methodName = (String)methods.next();
+      ImmutableList methodList = ImmutableList.copyOf(methodMap.get(methodName));
+
+      for(int i = 0; i < methodList.size(); ++i) {
+        ExecutableElement methodI = (ExecutableElement)methodList.get(i);
+
+        for(int j = i + 1; j < methodList.size(); ++j) {
+          ExecutableElement methodJ = (ExecutableElement)methodList.get(j);
+          if(elementUtils.overrides(methodJ, methodI, type)) {
+            overridden.add(methodI);
+          }
+        }
+      }
+    }
+
+    LinkedHashSet var11 = new LinkedHashSet(methodMap.values());
+    var11.removeAll(overridden);
+    return ImmutableSet.copyOf(var11);
+  }
+
+  private static void getLocalAndInheritedMethods(PackageElement pkg, TypeElement type, SetMultimap<String, ExecutableElement> methods) {
+    Iterator i$ = type.getInterfaces().iterator();
```

Comment:
Nope, just me being an idiot.  Didn't realize that following that went into a decompiled class file and copied it right out. 😖